armi/reactor/components/component.py

```diff
@@ -639,14 +636,49 @@ def getNumberDensity(self, nucName):
         number density : float
             number density in atoms/bn-cm.
         """
-        return self.p.numberDensities.get(nucName, 0.0)
+        i = np.where(self.p.nuclides == nucName.encode())[0]
+        if i.size > 0:
+            return self.p.numberDensities[i[0]]
+        else:
+            return 0.0
 
     def getNuclideNumberDensities(self, nucNames):
         """Return a list of number densities for the nuc names requested."""
-        return [self.p.numberDensities.get(nucName, 0.0) for nucName in nucNames]
+        if isinstance(nucNames, (list, tuple, np.ndarray)):
```

Comment:
Awww. A simple, one-line method becomes a 31-line method. That's a shame.

Reply:
I could write a simple version that is only a few lines, where we just reconstruct the full dictionary and then use the previous one-line function to get the number densities from the dictionary, but I would expect that to be slower.